### PR TITLE
[DOC] Contributing guide update to describe Custom Resource API Reference build

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation**
Update to the _Contributing Guide_ to describe what the _Custom Resource API Reference_ contains and how it is built

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: documentation/contributing/introduction.adoc [hunk 1]
```diff
@@ -80,4 +82,57 @@ In contrast to the _Deploying_ guide, this guide provides a reduced set of instr
...
+import io.strimzi.crdgenerator.annotations.DescriptionFile;
+----
+
+The tables in `appendix_crds.adoc` are built from `@Description` annotations in the Java files.
+
+Additional information is included by adding:
+
+. An `@DescriptionFile` annotation to the Java file
+. A corresponding description file (`.adoc`) in the `documentation/api/` folder
+. An `include:__DESCRIPTION-FILE-NAME__` reference in `appendix_crds.adoc`
```
Comment (scholzj) [inline documentation/contributing/introduction.adoc:117]: I think you should make it clear that this is generated automatically by the Maven build and does not have to be done manually.

File: documentation/contributing/introduction.adoc [hunk 2]
```diff
@@ -36,8 +36,10 @@ Assemblies, which usually encapsulate a feature or process, bring the related co
 An assembly is like a sub-section or chapter in a book.
 For example, the _Deploying Strimzi_ guide has a chapter called _Verifying the Strimzi deployment_, which is contained in its own assembly.
 
+.Documentation folder descriptions
 [source,options="nowrap",subs="+quotes"]
 ----
+api/                                 # Property descriptions for the Custom Resource API Reference
```
Comment (laidan6000) [inline documentation/contributing/introduction.adoc:42]:
```suggestion
api/                                 # Property descriptions for the Custom Resource API Reference section of the _Using Strimzi_ guide
```

File: documentation/contributing/introduction.adoc [hunk 3]
```diff
@@ -80,4 +82,57 @@ In contrast to the _Deploying_ guide, this guide provides a reduced set of instr
...
+
+The tables in `appendix_crds.adoc` are built from `@Description` annotations in the Java files.
+
+Additional information is included by adding:
+
+. An `@DescriptionFile` annotation to the Java file
+. A corresponding description file (`.adoc`) in the `documentation/api/` folder
+. An `include:__DESCRIPTION-FILE-NAME__` reference in `appendix_crds.adoc`
+
+For example, to add additional configuration for `KafkaUserQuotas`:
```
Comment (laidan6000) [inline documentation/contributing/introduction.adoc:119]: The `KafkaUserQuotas` object / resource?

File: documentation/contributing/introduction.adoc [hunk 4]
```diff
@@ -80,4 +82,57 @@ In contrast to the _Deploying_ guide, this guide provides a reduced set of instr
...
+. `appendix_crds.adoc` contains a reference to include the additional configuration description:
++
+[source,asciidoc,options="nowrap"]
+----
+### `KafkaUserQuotas` schema reference
+
+/include::../api/io.strimzi.api.kafka.model.KafkaUserQuotas.adoc[leveloffset=+1]
+----
+
+If you change any anything in the `api` module of the Java code, you must rebuild the _Custom Resource API Reference_ using a xref:make-tooling[make command].
```
Comment (laidan6000) [inline documentation/contributing/introduction.adoc:138]: Typo: any

You must rebuild the API Reference before submitting the PR?

**Review by laidan6000 — APPROVED** [review]
Great stuff. I added a few questions.

File: documentation/contributing/introduction.adoc (hunk 1 above)
Comment (PaulRMellor) [reply documentation/contributing/introduction.adoc:117]: > The `include:__DESCRIPTION-FILE-NAME__` reference is added automatically by the Maven build, so you do not need to add it manually.

File: documentation/contributing/introduction.adoc (hunk 3 above)
Comment (PaulRMellor) [reply documentation/contributing/introduction.adoc:121]: > `KafkaUserQuotas` custom resource

File: documentation/contributing/introduction.adoc (hunk 4 above)
Comment (PaulRMellor) [reply documentation/contributing/introduction.adoc:140]: :+1:  I added to the build section

> If you change the Strimzi API, you need to regenerate the API Reference before submitting your PR by running the following from the root:

**Review by scholzj — APPROVED** [review]

**Review by ppatierno — APPROVED** [review]

**Pull request merged**